doc/man1/kdf.pod

```diff
@@ -0,0 +1,152 @@
+
+=over 4
+
+=item B<key:string>
+
+Specifies the secret key as an alphanumeric string (use if the key contains
+printable characters only).
+The string length must conform to any restrictions of the KDF algorithm.
+A key must be specified for most KDF algorithms.
+
+=item B<hexkey:string>
+
+Specifies the secret key in hexadecimal form (two hex digits per byte).
+The key length must conform to any restrictions of the KDF algorithm.
+A key must be specified for most KDF algorithms.
+
+=item B<pass:string>
+
+Specifies the password as an alphanumeric string (use if the password contains
+printable characters only).
+The password must be specified for PBKDF2 and scrypt.
+
+=item B<hexpass:string>
+
+Specifies the password in hexadecimal form (two hex digits per byte).
+The password must be specified for PBKDF2 and scrypt.
+
+=item B<digest:string>
+
+Specifies the name of a digest as an alphanumeric string.
+To see the list of supported digests, use the command I<list -digest-commands>.
+
+=back
+
+=item B<kdf_name>
+
+Specifies the name of a supported KDF algorithm which will be used.
+The supported algorithms are TLS1-PRF, HKDF, SSKDF, PBKDF2, SSHKDF and id-scrypt.
+
+=back
+
+=head1 EXAMPLES
+
+=head2 Use TLS1-PRF to create a hex-encoded derived key from a secret key and seed:
+
+C<openssl kdf -keylen 16 -kdfopt digest:SHA256 -kdfopt key:secret -kdfopt seed:seed TLS1-PRF>
+
+=head2 Use HKDF to create a hex-encoded derived key from a secret key, salt and info:
+
+C<openssl kdf -keylen 10 -kdfopt digest:SHA256 -kdfopt key:secret -kdfopt salt:salt -kdfopt info:label HKDF>
+
+=head2 Use SSKDF with KMAC to create a hex-encoded derived key from a secret key, salt and info:
+
+C<openssl kdf -keylen 64 -kdfopt mac:KMAC128 -kdfopt maclen:20 -kdfopt hexkey:b74a149a161545 -kdfopt hexinfo:348a37a2 -kdfopt hexsalt:3638271ccd68a2 SSKDF>
+
+=head2 Use SSKDF with HMAC to create a hex-encoded derived key from a secret key, salt and info:
+
+C<openssl kdf -keylen 16 -kdfopt mac:HMAC -kdfopt digest:SHA256 -kdfopt hexkey:b74a149a -kdfopt hexinfo:348a37a2 -kdfopt hexsalt:3638271c SSKDF>
+
+=head2 Use SSKDF with Hash to create a hex-encoded derived key from a secret key, salt and info:
+
+C<openssl kdf -keylen 14 -kdfopt digest:SHA256 -kdfopt hexkey:6dbdc23f045488 -kdfopt hexinfo:a1b2c3d4 SSKDF>
+
+=head2 Use SSHKDF to create a hex-encoded derived key from a secret key, hash and session_id:
+
+C<openssl kdf -keylen 16 -kdfopt md:SHA256 -kdfopt hexkey:0102030405 -kdfopt hexxcghash:06090A -kdfopt hexsession_id:01020304 -kdfopt type:A SSHKDF>
+
+=head2 Use PBKDF2 to create a hex-encoded derived key from a password and salt:
+
+C<openssl kdf -keylen 32 -kdfopt digest:SHA256 -kdfopt pass:password -kdfopt salt:salt -kdfopt iter:2 PBKDF2>
+
+=head2 Use scrypt to create a hex-encoded derived key from a password and salt:
+
+C<openssl kdf -keylen 64 -kdfopt pass:password -kdfopt salt:NaCl -kdfopt N:1024 -kdfopt r:8 -kdfopt p:16 -kdfopt maxmem_bytes:10485760 id-scrypt>
+
```

Comment:
I'm sorry, I thoroughly dislike what you've done here.  Not sure what's wrong with the following?

```
=head1 EXAMPLES

Use TLS1-PRF to create a hex-encoded derived key from a secret key and seed:

    openssl kdf -keylen 16 -kdfopt digest:SHA256 -kdfopt key:secret \
                -kdfopt seed:seed TLS1-PRF

Use HKDF to create a hex-encoded derived key from a secret key, salt and info:

    openssl kdf -keylen 10 -kdfopt digest:SHA256 -kdfopt key:secret \
                -kdfopt salt:salt -kdfopt info:label HKDF

Use SSKDF with KMAC to create a hex-encoded derived key from a secret key, salt and info:

    openssl kdf -keylen 64 -kdfopt mac:KMAC128 -kdfopt maclen:20 \
                -kdfopt hexkey:b74a149a161545 -kdfopt hexinfo:348a37a2 \
                -kdfopt hexsalt:3638271ccd68a2 SSKDF

Use SSKDF with HMAC to create a hex-encoded derived key from a secret key, salt and info:

    openssl kdf -keylen 16 -kdfopt mac:HMAC -kdfopt digest:SHA256 \
                -kdfopt hexkey:b74a149a -kdfopt hexinfo:348a37a2 \
                -kdfopt hexsalt:3638271c SSKDF

Use SSKDF with Hash to create a hex-encoded derived key from a secret key, salt and info:

    openssl kdf -keylen 14 -kdfopt digest:SHA256 \
                -kdfopt hexkey:6dbdc23f045488 \
                -kdfopt hexinfo:a1b2c3d4 SSKDF

Use SSHKDF to create a hex-encoded derived key from a secret key, hash and session_id:

    openssl kdf -keylen 16 -kdfopt md:SHA256 \
                -kdfopt hexkey:0102030405 \
                -kdfopt hexxcghash:06090A \
                -kdfopt hexsession_id:01020304 \
                -kdfopt type:A SSHKDF>

Use PBKDF2 to create a hex-encoded derived key from a password and salt:

    openssl kdf -keylen 32 -kdfopt digest:SHA256 -kdfopt pass:password \
                -kdfopt salt:salt -kdfopt iter:2 PBKDF2

Use scrypt to create a hex-encoded derived key from a password and salt:

    openssl kdf -keylen 64 -kdfopt pass:password -kdfopt salt:NaCl \
                -kdfopt N:1024 -kdfopt r:8 -kdfopt p:16 \
                -kdfopt maxmem_bytes:10485760 id-scrypt
```

Reply:
The output from perldoc looks like cat vomit.. Will change it :)

Reply:
Huh? Not sure I understand what you mean with "cat vomit"

Reply:
It looks ok with your latest format, it is a pity it doesnt copy/paste very nicely to the commandline for testing purposes.